android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -830,48 +846,83 @@ public void run() {
                 }
             });
         } else {
-            // Make new network specifier
-            final WifiNetworkSpecifier.Builder builder = new WifiNetworkSpecifier.Builder();
-            // set ssid
-            builder.setSsid(ssid);
-            // set security
-            if (security != null && security.toUpperCase().equals("WPA")) {
-                builder.setWpa2Passphrase(password);
-            } else if (security != null && security.toUpperCase().equals("WEP")) {
-                // WEP is not supported
-                poResult.error("Error", "WEP is not supported for Android SDK " + Build.VERSION.SDK_INT, "");
+            // error if WEP security, since not supported
+            if (security != null && security.toUpperCase().equals("WEP")) {
+                handler.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        poResult.error("Error", "WEP is not supported for Android SDK " + Build.VERSION.SDK_INT, "");
+                    }
+                });
+                return;
             }
 
-            final NetworkRequest networkRequest = new NetworkRequest.Builder()
-                    .addTransportType(NetworkCapabilities.TRANSPORT_WIFI)
-                    .removeCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET)
-                    .setNetworkSpecifier(builder.build())
-                    .build();
-
-            final ConnectivityManager connectivityManager = (ConnectivityManager) moContext.getSystemService(Context.CONNECTIVITY_SERVICE);
+            if (withInternet != null && withInternet) {
+                // create network suggestion
+                final WifiNetworkSuggestion.Builder builder = new WifiNetworkSuggestion.Builder();
+                // set ssid
+                builder.setSsid(ssid);
+                // set password
+                if (security != null && security.toUpperCase().equals("WPA")) {
+                    builder.setWpa2Passphrase(password);
+                }
+                //builder.setIsAppInteractionRequired(true);
+                final WifiNetworkSuggestion suggestion = builder.build();
 
+                networkSuggestions = new ArrayList<>();
```

Comment:
If `networkSuggestions` already exist we may then lose what was inside. I believe this can happen only if one calls `connect` again without calling a `disconnect` first.
I would propose that in such a case we cleanup the connections (so call `                moWiFi.removeNetworkSuggestions(networkSuggestions);`) or put a mechanism in place to prevent asking for a `connect` while one is still "active".

Reply:
i agree with you, what should we go with ?  I think we should with "clean up connection" mechanism. what is your vote?

Reply:
added "clean up connection code" in e3ab29b, resolving this. Open it if you have different views.

Reply:
sounds good to clean up, so we do not leave the system in a different state. We may in case do the failure in case of an already ongoing connection later on if needed!